src/Diagnostics/FxCop/SDL.Analyzer.Common/CSharp/Shared/SyntaxNodeHelper.cs

```diff
@@ -0,0 +1,323 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Symbols;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+
+namespace SDL.Analyzer.Common
+{
+    public sealed class CSharpSyntaxNodeHelper : SyntaxNodeHelper
+    {
+        private static CSharpSyntaxNodeHelper instance = new CSharpSyntaxNodeHelper();
+
+        public static CSharpSyntaxNodeHelper Default { get { return instance; } }
+
+        private CSharpSyntaxNodeHelper()
+        {
+        }
+
+        public override bool ContainsMethodCall(SyntaxNode node, Func<string, bool> predicate)
+        {
+            if (node == null)
+            {
+                return false;
+            }
+
+            return node.DescendantNodesAndSelf().OfType<InvocationExpressionSyntax>().Any(
+                (InvocationExpressionSyntax child) =>
+                {
+                    return child.DescendantNodesAndSelf().OfType<IdentifierNameSyntax>().Any(name => predicate(name.Identifier.ValueText));
+                });
+        }
+
+        public override IMethodSymbol GetCallerMethodSymbol(SyntaxNode node, SemanticModel semanticModel)
+        {
+            if (node == null)
+            {
+                return null;
+            }
+
+            MethodDeclarationSyntax declaration = node.AncestorsAndSelf().OfType<MethodDeclarationSyntax>().FirstOrDefault();           
+            if (declaration != null)
+            {
+                return semanticModel.GetDeclaredSymbol(declaration);
+            }
+
+            ConstructorDeclarationSyntax contructor = node.AncestorsAndSelf().OfType<ConstructorDeclarationSyntax>().FirstOrDefault();
+            if (contructor != null)
+            {
+                return semanticModel.GetDeclaredSymbol(contructor);
+            }
+
+            return null;
+        }
+
+        public override ITypeSymbol GetEnclosingTypeSymbol(SyntaxNode node, SemanticModel semanticModel)
```

Comment:
This should not be required as well. Analysis within code blocks will have the enclosing member symbol (see the above comment), and you can fetch the containing type symbol from there.